### PR TITLE
python312Packages.minari: 0.4.3 -> 0.5.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

- **python312Packages.mktestdocs: init at 0.2.1**
- **python312Packages.minari: 0.4.3 -> 0.5.0**

Changelog: https://github.com/Farama-Foundation/Minari/releases/tag/v0.5.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 338197` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>python311Packages.minari</li>
    <li>python311Packages.minari.dist</li>
    <li>python311Packages.mktestdocs</li>
    <li>python311Packages.mktestdocs.dist</li>
    <li>python312Packages.mktestdocs</li>
    <li>python312Packages.mktestdocs.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 338197` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>python311Packages.minari</li>
    <li>python311Packages.minari.dist</li>
    <li>python311Packages.mktestdocs</li>
    <li>python311Packages.mktestdocs.dist</li>
    <li>python312Packages.mktestdocs</li>
    <li>python312Packages.mktestdocs.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 338197` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>minari</li>
    <li>minari.dist</li>
    <li>python311Packages.minari</li>
    <li>python311Packages.minari.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.mktestdocs</li>
    <li>python311Packages.mktestdocs.dist</li>
    <li>python312Packages.mktestdocs</li>
    <li>python312Packages.mktestdocs.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 338197` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>python311Packages.minari</li>
    <li>python311Packages.minari.dist</li>
    <li>python311Packages.mktestdocs</li>
    <li>python311Packages.mktestdocs.dist</li>
    <li>python312Packages.mktestdocs</li>
    <li>python312Packages.mktestdocs.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/minari/default.nix [hunk 1]
```diff
@@ -33,56 +39,74 @@ buildPythonPackage rec {
...
+  nativeCheckInputs = [
+    google-cloud-storage
+    h5py
+    jax
+    jaxlib
+    mktestdocs
+    pyarrow
+    pytestCheckHook
+    tqdm
+  ];
```
[the remark this answers is not in the document]
Comment (SuperSandro2000) [reply pkgs/development/python-modules/minari/default.nix:83]: you can make this convenient like https://github.com/NixOS/nixpkgs/blob/21576aba43e604a95cc45c25248f39f3989ebe36/pkgs/by-name/mu/music-assistant/package.nix#L108

[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/development/python-modules/minari/default.nix]: Good idea, done

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 338197` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>python311Packages.minari</li>
    <li>python311Packages.minari.dist</li>
    <li>python311Packages.mktestdocs</li>
    <li>python311Packages.mktestdocs.dist</li>
    <li>python312Packages.mktestdocs</li>
    <li>python312Packages.mktestdocs.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]

**Pull request merged**